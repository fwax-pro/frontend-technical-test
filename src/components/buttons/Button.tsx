import type { FC } from "react"
import styled from "styled-components"

interface ButtonProps {
    label?: String
    children?: any
    onClick?: () => void
}

export const Button: FC<ButtonProps> = ({
    children,
    onClick
}) => (
    <InnerButton onClick={onClick}>
        { children }
    </InnerButton>
);

const InnerButton = styled.button`
    background-color: transparent;
    border: none;
    color: ${({ theme }) => theme.colors.secondary};
    text-align: center;
    border-radius: 50%;
    margin-right: 1rem;
    display: flex;
    justify-content: center;
    align-items: center;
    width: 3rem;
    height: 3rem;    
`;
