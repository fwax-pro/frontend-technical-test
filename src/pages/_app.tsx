import type { AppProps } from 'next/app'
import { LayoutGroup } from "framer-motion"
import { ThemeProvider, DefaultTheme } from 'styled-components'

import initAxios from '../libs/axios'

import GlobalStyle from '../components/globalstyles'

import '../styles/static.css'


initAxios();

export const theme: DefaultTheme = {
  colors: {
    primary: '#ff6e14',
    secondary: '#fff',
  },
}

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <ThemeProvider theme={theme}>
      <LayoutGroup>
        <GlobalStyle />
        <Component {...pageProps} />
      </LayoutGroup>
    </ThemeProvider>
  )
}

export default MyApp
