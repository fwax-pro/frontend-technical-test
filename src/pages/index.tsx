import { FC } from 'react'
import Link from 'next/link'
import { Layout } from '../components/layout'
import { InnerLoginLink, MainCentered, MainTitle } from '../components/sharedstyles'


const Home: FC = () => {
  return (
    <Layout>
      <MainCentered>
        <MainTitle>lamessagerie</MainTitle>
        <Link href="/messages">
          <InnerLoginLink>
            connecter
          </InnerLoginLink>
        </Link>
      </MainCentered>
    </Layout>
  )
}


export default Home
