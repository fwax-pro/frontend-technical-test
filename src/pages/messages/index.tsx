import { FC } from "react"
import Image from "next/image"
import { useRouter } from 'next/router'
import axios from "axios"

import { Conversations } from '../../components/conversation'
import { Header, Layout, Main } from '../../components/layout'
import { Avatar, Title } from '../../components/sharedstyles'

import { getLoggedUserId } from '../../utils/index'
import { GetStaticProps } from "next"
import { Conversation } from "../../types/conversation"

interface ConversationsPageProps {
    conversations: Conversation[]
    loggedUserId: number
}

const ConversationsPage: FC<ConversationsPageProps> = ({ loggedUserId, conversations }) => {
    const router = useRouter();

    return (
        <Layout>
            <Header>
                <Avatar>
                    <Image 
                        src={`${router.basePath}/avatars/user-${loggedUserId}.png`} 
                        width={64} 
                        height={64}
                        alt='avatar user'
                    />
                </Avatar>
                <Title>Discussions</Title>
            </Header>
            <Main>
                <Conversations conversations={conversations} loggedUserId={loggedUserId} />
            </Main>
        </Layout>
    )
}

export const getStaticProps:GetStaticProps  = async () => {
    const loggedUserId = getLoggedUserId();
    const res = await axios.get(`/conversations/${loggedUserId}`);

    return { props: {loggedUserId, conversations: res.data } }
}

export default ConversationsPage;
