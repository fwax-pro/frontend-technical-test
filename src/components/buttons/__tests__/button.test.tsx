import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { Button } from '../Button'
import { theme } from "../../../pages/_app"

describe('components Button', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(
        <ThemeProvider theme={theme}>
            <Button/>
        </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
