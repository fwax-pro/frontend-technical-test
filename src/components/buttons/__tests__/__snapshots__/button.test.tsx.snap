// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Button should match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bczRLJ dyRODB"
  />
</DocumentFragment>
`;
